packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -107,17 +114,26 @@ class ConfigChainBuilder {
 
     this.resolvedConfigs.push(loc);
 
-    let content = fs.readFileSync(loc, "utf8");
     let options;
-
-    try {
-      options = jsonCache[content] = jsonCache[content] || json.parse(content);
-      if (key) options = options[key];
-    } catch (err) {
-      err.message = `${loc}: Error while parsing JSON - ${err.message}`;
-      throw err;
+    if (path.extname(loc) === ".js") {
+      try {
+        options = require(loc);
```

Comment:
Might be worth checking for a `.default` prop on the returned object. Someone could be using `babel-register` and have `export default {}` for `.babelrc.js`